src/vanilla/DanModel/JavaEnum.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace AutoRest.Java.DanModel
+{
+    public class JavaEnum
+    {
+        public const string RelativePackage = "models";
+
+        private readonly IList<JavaEnumValue> values;
+
+        public JavaEnum(string headerCommentText, string package, string enumName)
+        {
+            HeaderCommentText = headerCommentText;
+            Package = package;
+            EnumName = enumName;
+            values = new List<JavaEnumValue>();
+        }
+
+        public JavaEnum AddValue(string valueName, string valueString)
```

Comment:
I think these models should be fully immutable.

Reply:
I think that's a reasonable request. I'll do that.